packages/glob/__tests__/internal-globber.test.ts

```diff
@@ -734,6 +734,26 @@ describe('globber', () => {
     ])
   })
 
+  it('ignores hidden files when excludeHiddenFiles is set', async () => {
+    // Create the following layout:
+    //   <root>
+    //   <root>/.emptyFolder
+    //   <root>/.file
+    //   <root>/.folder
+    //   <root>/.folder/file
+    const root = path.join(getTestTemp(), 'ignores-hidden-files')
+    await createHiddenDirectory(path.join(root, '.emptyFolder'))
+    await createHiddenDirectory(path.join(root, '.folder'))
+    await createHiddenFile(path.join(root, '.file'), 'test .file content')
+    await fs.writeFile(
```

Comment:
I'm a little confused about the behaviour here.

I'm not sure if the behaviour for the following globs matches what's intended
* `ignores-hidden-files/.folder` - returns nothing
* `ignores-hidden-files/.folder/*` - returns `file`
* `ignores-hidden-files/.folder/file` - returns `file`